agent/src/AgentGlobalState.ts

```diff
@@ -11,7 +13,10 @@ import * as vscode_shim from './vscode-shim'
 export class AgentGlobalState implements vscode.Memento {
     private globalStorage = new Map<string, any>()
 
-    constructor() {
+    constructor(private path?: string) {
```

Comment:
Let's make this a directory to 1) ensure we have a consistent filename and 2) making it possible to write multiple files (for example, sqlite lockfiles)

```suggestion
    constructor(private dir?: string) {
```